server/modules/survey/manager/surveyManager.js

```diff
@@ -271,17 +272,24 @@ export const publishSurveyProps = async (surveyId, langsDeleted, client = db) =>
     }
   })
 
-export const { updateSurveyDependencyGraphs } = SurveyRepository
+export const { removeSurveyTemporaryFlag, updateSurveyDependencyGraphs } = SurveyRepository
 
 // ====== DELETE
-export const deleteSurvey = async (surveyId) =>
-  db.tx(async (t) => {
-    await Promise.all([
-      UserRepository.deleteUsersPrefsSurvey(surveyId, t),
-      SurveyRepository.dropSurveySchema(surveyId, t),
-      SchemaRdbRepository.dropSchema(surveyId, t),
-      SurveyRepository.deleteSurvey(surveyId, t),
-    ])
+export const deleteSurvey = async (surveyId, { deleteUserPrefs = true } = {}, client = db) =>
+  client.tx(async (t) => {
+    if (deleteUserPrefs) {
+      await UserRepository.deleteUsersPrefsSurvey(surveyId, t)
+    }
+    await SurveyRepository.deleteSurvey(surveyId, t)
```

Comment:
maybe is better delete the schema before the survey?
I don't know

Reply:
I've put it in the end because it's under transaction, if something goes wrong in the first calls the transaction will be rolledback and the survey schema will still be there!